modules/PROGENY.py

```diff
@@ -1,234 +1,140 @@
-    'Androgen': 'R-HSA-8940973:RUNX2 regulates osteoblast differentiation',
-    'EGFR': 'R-HSA-8856828:Clathrin-mediated endocytosis',
-    'Estrogen': 'R-HSA-8939902:Regulation of RUNX2 expression and activity',
-    'Hypoxia': 'R-HSA-123456: XXX Pathway desc to be updated',
-    'JAK.STAT': 'R-HSA-6785807:Interleukin-4 and 13 signaling',
-    'MAPK': 'R-HSA-2559580:Oxidative Stress Induced Senescence',
-    'NFkB': 'R-HSA-9020702:Interleukin-1 signaling',
-    'PI3K': 'R-HSA-8853659:RET signaling',
-    'TGFb': 'R-HSA-2173788:Downregulation of TGF-beta receptor signaling',
-    'TNFa': 'R-HSA-5357956:TNFR1-induced NFkappaB signaling pathway',
-    'Trail': 'R-HSA-123456:XXX Pathway desc to be updated',
-    'VEGF': 'R-HSA-1234158:Regulation of gene expression by Hypoxia-inducible Factor',
-    'WNT': 'R-HSA-381340:Transcriptional regulation of white adipocyte differentiation',
-    'p53': 'R-HSA-2559580:Oxidative Stress Induced Senescence'
-}
-
-# === These symbols are secondary/generic/typo that need updating ===
-PROGENY_SYMBOL_MAPPING = {
-    'NKFB1': 'NFKB1',
-    'MAPK2K1': 'PRKMK1',
-    'PI3K(Class1)': 'PIK3CA',
-    'VEGFR': 'KDR',
-    'BCL-W': 'BCL2L2',
-    'BCL-XL': 'BCL2L1'
-}
-
-PROGENY_DATABASE_ID='PROGENY'
-PROGENY_VERSION='2018.04'
-PROGENY_PUBLICATION="http://europepmc.org/abstract/MED/29295995"
-
-class PROGENY:
-    def __init__(self):
-        self.evidence_strings = list()
-        self.symbols = {}
-        self.logger = logging.getLogger(__name__)
-
-    def process_progeny(self):
-        gene_parser = GeneParser()
-        gene_parser._get_hgnc_data_from_json()
-
-        self.symbols = gene_parser.genes
-        self.build_evidence()
-        self.write_evidence()
-
-    def build_evidence(self, filename=Config.PROGENY_FILENAME):
-
-        now = datetime.datetime.now()
-        # *** Build evidence.provenance_type object ***
-        provenance_type = evidence_core.BaseProvenance_Type(
-            database=evidence_core.BaseDatabase(
-                id=PROGENY_DATABASE_ID,
-                version=PROGENY_VERSION,
-                dbxref=evidence_core.BaseDbxref(url="https://saezlab.github.io/progeny/", id="PROGENY Pathway association analysis of TCGA tumor types", version=PROGENY_VERSION)),
-            literature=evidence_core.BaseLiterature(
-                references=[evidence_core.Single_Lit_Reference(lit_id=PROGENY_PUBLICATION)]
-            )
+import json
+from pyspark import SparkContext, SparkFiles
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import *
+from pyspark.sql.types import *
+
+class progenyEvidenceGenerator():
+    def __init__(self, inputFile, mappingStep):
+        # Create spark session     
+        self.spark = SparkSession.builder \
+                .appName('progeny') \
+                .getOrCreate()
+
+        # Initialize mapping variables
+        self.mappingStep = mappingStep
+    
+        # Initialize input files
+        self.inputFile = inputFile
```

Comment:
Same as above: does this need to be an object attribute? It could be passed to directly to `writeEvidenceFromSource()` which is where it's used